src/Symfony/Component/Translation/Translator.php

```diff
@@ -37,54 +37,36 @@ class Translator implements TranslatorInterface, TranslatorBagInterface, LocaleA
      */
     protected $catalogues = [];
 
-    /**
-     * @var string
-     */
-    private $locale;
+    private string $locale;
 
     /**
-     * @var array
+     * @var string[]
      */
-    private $fallbackLocales = [];
+    private array $fallbackLocales = [];
 
     /**
      * @var LoaderInterface[]
      */
-    private $loaders = [];
+    private array $loaders = [];
 
-    /**
-     * @var array
-     */
-    private $resources = [];
+    private array $resources = [];
 
-    /**
-     * @var MessageFormatterInterface
-     */
-    private $formatter;
+    private MessageFormatterInterface $formatter;
 
     /**
      * @var string
      */
-    private $cacheDir;
+    private ?string $cacheDir;
 
-    /**
-     * @var bool
-     */
-    private $debug;
+    private bool $debug;
 
-    private $cacheVary;
+    private array $cacheVary;
 
-    /**
-     * @var ConfigCacheFactoryInterface|null
-     */
-    private $configCacheFactory;
+    private ?ConfigCacheFactoryInterface $configCacheFactory;
 
-    /**
-     * @var array|null
-     */
-    private $parentLocales;
+    private ?array $parentLocales;
```

Comment:
should not be nullable, as it was null before only before initializing it.

Reply:
Removed nullable in 1516fd39114f2f19942a1ba4cf80ec868d46e181

Actually, `$parentLocales` does not get initialized in the constructor, so although `computeFallbackLocales` has an `isset` check, are we sure `initializeCatalogue` does not crash because of that?

Reply:
Calling `isset()` on uninitialized properties is safe. Same goes for `??` and `??=`.